### PR TITLE
 Fix deadlock when installCharts is stuck in a loop 

### PR DESCRIPTION
**Pull request by ibuildthecloud**

### CODE REVIEW

**Comment by kinarashah** [comment]
https://github.com/rancher/rancher/issues/34835

**Review by kinarashah — APPROVED** [review]
Validated using `kinarashah/rancher:v2.6.1-rc3` with the fix.

**Review by nickgerace — APPROVED** [review]

File: pkg/catalogv2/system/system.go [hunk 1]
```diff
@@ -120,7 +120,10 @@ func (m *Manager) onTrigger(_ string, obj *catalog.ClusterRepo) (*catalog.Cluste
 		return obj, nil
 	}
 
-	m.trigger <- struct{}{}
+	select {
+	case m.trigger <- struct{}{}:
+	default:
+	}
```
Comment (thedadams) [inline pkg/catalogv2/system/system.go:126]: Do we want to error or re-enqueue in the `default` case to ensure this eventually gets reconciled?

Comment (ibuildthecloud) [reply pkg/catalogv2/system/system.go:126]: I don't think so because if this happens the queue already has one item in it.  That chan is of length 1.

Comment (cbron) [reply pkg/catalogv2/system/system.go:126]: Hmm, thats an interesting point. If the catalog is rancher-charts and its blocked, then default executes and nothing happens until it specifically gets changed/refreshed again.

**Pull request merged**